azkaban-exec-server/src/main/java/azkaban/execapp/ProjectCacheCleaner.java

```diff
@@ -155,7 +155,7 @@ private void deleteLeastRecentlyUsedProjects(long sizeToFreeInBytes,
     final Set<File> projectDirsToDelete = new HashSet<>();
 
     for (final ProjectDirectoryMetadata proj : projectDirMetadataList) {
-      if (sizeToFreeInBytes > 0) {
+      if (sizeToFreeInBytes > 0 || projectDirsToDelete.size() < 5) {
```

Comment:
When you graduate this from MVP, please make this configurable (with default value of 5? but why 5 would still be a question :) )

Reply:
...or "if". 
And yes, of course, as mentioned above:
> This is a concept demonstration only. A more robust implementation would probably allow max # of dirs cleaned up per invocation to be parameterised, with some reasonable default provided as well. (Between two and five would be a reasonable default to guarantee that we are deleting more than we’re adding, on average.) Tests, too, would need to be added.